lib/src/connector/connection.cc

```diff
@@ -234,11 +234,10 @@ void Connection::connect(int max_connect_attempts)
                 // Connection attempt failed, next try should be against a failover broker.
                 switchWsUri();
                 // Randomly adjust the interval slightly to help calm
-                // a thundering herd
                 doSleep(connection_backoff_ms_ + dist(engine));
                 connectAndWait();
                 if (try_again && !got_max_backoff) {
-                    connection_backoff_ms_ *= CONNECTION_BACKOFF_MULTIPLIER;
+                    connection_backoff_ms_ *= static_cast<int>(dist(engine) / 1000 + CONNECTION_BACKOFF_MULTIPLIER);
```

Comment:
I'd like a comment clarifying the back off strategy, as it took me a minute to think through. Looks like the effect will be an exponential backoff that varies between a 1.5-2.5x multiplier, that retains the 32s limit.